src/base/preferences.cpp

```diff
@@ -718,6 +718,16 @@ void Preferences::useSystemIconTheme(bool enabled)
 }
 #endif
 
+bool Preferences::useSystemFileIcons() const
```

Comment:
>IMO, our current approach is best suited for applications that use context menus for configurations, but not the central options dialog. I will make the requested changes, through.

Agree. The current (still not used everywhere) approach was not as good as I wanted it. The same old approach were bad. Apparently, I moved it from one extreme to the other. I often ponder how to make it better.